### DISCUSSION TITLE
Change the SourceTimestamp of a DataValue

### DISCUSSION THREAD

**Discussion by maki1-1** [FreeOpcUa/opcua-asyncio#591] [Q&A]
Hi,

I think this is a minor question, however I stuck.

I have a variable with the following DataValue:
`DataValue(Value:Variant(val:36,type:VariantType.Int64), StatusCode:StatusCode(Good), SourceTimestamp:2021-06-04 14:22:47.425125)`

My client (free opc-ua gui client) lists the SourceTimestamp in UTC, which is 2 hours earlier than my Timezone. I figured that I can fix it with 

`.set_data_value()`

However, I´m confused with the syntax in the definition: What does self mean?

`def set_datavalue(self, value, varianttype=None):`

Also, the ServerTimestamp is None. How can that be changed? (If needed)
Any help is welcome!

**Comment by AndreasHeine** [comment]
https://reference.opcfoundation.org/v104/Core/docs/Part4/7.7.1/
always use UTC only for the UI itself use Timezones!

if you create an DataValue you can set the servertimestamp aswell!
its normaly set automaticly!?... which version your at?

"self" is like "this" in js ^^
https://www.geeksforgeeks.org/self-in-python-class/

**Reply by maki1-1** [reply]
Thanks for the fast reply.

I´m using python 3.8 and opcua (however, not asyncio so far, which may be why its not set automatically).

To make things clear to me:
self = the variable of interest? in my case CO2_Concentration
Value = the actual value of the variable (Result_upload[1])

So it would be 

Variable.set_data_value(Variable, Value,)

but where do I then change the timezone? Sorry, I´m really confused.


I´m asking this because an opc client (provided by a biotech company) showed BAD values. Thus, I think it is because the original SourceTimestamp of the variable is not the same as in the client.

**Reply by AndreasHeine** [reply]
you have to use the utc time in datavalues! its part of opc ua spec. and all it systems use utc to have a common timebase -> timezones are only for humans which need light from 6 am - 6 pm 😉
